core/src/main/java/com/linecorp/armeria/internal/common/RequestContextUtil.java

```diff
@@ -225,20 +225,24 @@ public static void pop(RequestContext current, @Nullable RequestContext toRestor
     public static SafeCloseable invokeHookAndPop(RequestContext current, @Nullable RequestContext toRestore) {
         requireNonNull(current, "current");
 
-        final SafeCloseable closeable = invokeHook(current);
+        final AutoCloseable closeable = invokeHook(current);
         if (closeable == null) {
             return () -> requestContextStorage.pop(current, toRestore);
         } else {
             return () -> {
-                closeable.close();
+                try {
+                    closeable.close();
+                } catch (Throwable t) {
+                    logger.warn("{} Unexpected exception while closing RequestContext.hook().", current, t);
```

Comment:
Previously, we used `SafeCloseable` that does not throw any exception. I changed it to `AutoCloseable` because of https://github.com/line/armeria/pull/4082#discussion_r824690853.